x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/policy_list/policy_retention_schedule/policy_retention_schedule.tsx

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingContent,
+  EuiPanel,
+  EuiText,
+  EuiButton,
+  EuiButtonIcon,
+  EuiSpacer,
+  EuiToolTip,
+  EuiCallOut,
+} from '@elastic/eui';
+
+import { useAppDependencies } from '../../../../index';
+import { PolicyUpdateRetentionProvider } from '../../../../components';
+
+interface Props {
+  retentionSettings: {
+    retentionSchedule: string;
+  };
+  onRetentionScheduleUpdated: () => void;
+  isLoading: boolean;
+  error: any;
+}
+
+export const PolicyRetentionSchedule: React.FunctionComponent<Props> = ({
+  retentionSettings,
+  onRetentionScheduleUpdated,
+  isLoading,
+  error,
+}) => {
+  const {
+    core: { i18n },
+  } = useAppDependencies();
+
+  const { FormattedMessage } = i18n;
+
+  if (isLoading) {
+    return (
+      <Fragment>
+        <EuiPanel>
+          <EuiLoadingContent lines={1} />
+        </EuiPanel>
+        <EuiSpacer />
+      </Fragment>
+    );
+  }
+
+  if (error) {
+    return (
+      <Fragment>
+        <EuiCallOut
+          title={
+            <FormattedMessage
+              id="xpack.snapshotRestore.policyRetentionSchedulePanel.errorFetchingRetentionScheduleTitle"
+              defaultMessage="Error fetching retention schedule"
+            />
+          }
+          color="danger"
+          iconType="alert"
+        >
+          {error.data && error.data.message ? <p>{error.data.message}</p> : null}
+          <EuiButton iconType="refresh" color="danger" onClick={onRetentionScheduleUpdated}>
+            <FormattedMessage
+              id="xpack.snapshotRestore.policyRetentionSchedulePanel.errorFetchingRetentionScheduleReloadButtonLabel"
+              defaultMessage="Reload"
+            />
+          </EuiButton>
+        </EuiCallOut>
+        <EuiSpacer />
+      </Fragment>
+    );
+  }
+
+  if (retentionSettings && retentionSettings.retentionSchedule) {
```

Comment:
This is quite a large `if..else` statement - would it be possible to compress it somehow so that the overarching logic is communicated more easily for readers?

Reply:
thanks for pointing this out. i actually had some duplicated logic in here. i cleaned it up, so hopefully it's a little easier to follow now.